java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageMetadataResult.java

```diff
@@ -42,25 +42,39 @@
     this.message = message;
   }
 
+  public static MessageMetadataResult create(ByteBuffer buffer) {
+    return new MessageMetadataResult(-1, buffer, Message.getRootAsMessage(buffer));
+  }
 
   /**
    * Get the length of the message metadata in bytes, not including the body length.
    *
    * @return number of bytes in the message metadata buffer.
+   *
+   * @deprecated unused.
    */
+  @Deprecated
```

Comment:
I use this downstream in Spark, can you not deprecate it? I found it useful to not have to rely on calculating the ByteBuffer size that is created in Arrow when reading from a `ReadChannel`.

Reply:
great, thanks!